### PR TITLE
Add more linters

### PR DESCRIPTION
**Pull request by lucacome**
Adding more linters and fixes for the issues found.

### CODE REVIEW

**Review by soneillf5 — APPROVED** [review]

File: client/nginx.go [hunk 1]
```diff
@@ -43,25 +45,31 @@ func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient,
 
 // GetStubStats fetches the stub_status metrics.
 func (client *NginxClient) GetStubStats() (*StubStats, error) {
-	resp, err := client.httpClient.Get(client.apiEndpoint)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, client.apiEndpoint, nil)
```
Comment (pleshakov) [inline client/nginx.go:51]: what is the rationale for using `NewRequestWithContext`?

**Review by ciarams87 — APPROVED** [review]

File: client/nginx.go [hunk 2]
```diff
@@ -43,25 +45,31 @@ func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient,
 
 // GetStubStats fetches the stub_status metrics.
 func (client *NginxClient) GetStubStats() (*StubStats, error) {
-	resp, err := client.httpClient.Get(client.apiEndpoint)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
[the remark this answers is not in the document]
Comment (lucacome) [reply client/nginx.go]: passed `client.httpClient.Timeout`

File: client/nginx.go [hunk 3]
```diff
@@ -43,25 +45,31 @@ func NewNginxClient(httpClient *http.Client, apiEndpoint string) (*NginxClient,
...
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, client.apiEndpoint, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a get request: %w", err)
+	}
+	resp, err := client.httpClient.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get %v: %v", client.apiEndpoint, err)
+		return nil, fmt.Errorf("failed to get %v: %w", client.apiEndpoint, err)
 	}
-	defer resp.Body.Close()
```
[the remark this answers is not in the document]
Comment (lucacome) [reply client/nginx.go:50]: added back

File: client/nginx.go (hunk 1 above)
Comment (lucacome) [reply client/nginx.go:51]: Coming from the new linter

> You should use noctx if sending http request in your library. Passing context.Context enables library user to cancel http request, getting trace information and so on.

File: client/nginx.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucacome) [reply client/nginx.go]: I just realized that the timeout flag can be set to `0`, with this change it means that `0` will be passed to the context making it cancel it immediately. I need to think about this, any ideas?

[the remark this answers is not in the document]
Comment (pleshakov) [reply client/nginx.go]: would it make sense to use a context without the timeout in this case?  `context.TODO()`

File: client/nginx.go (hunk 1 above)
Comment (pleshakov) [reply client/nginx.go:50]: I don't think we use `cancel http request, getting trace information` in our Prometheus exporter, so not fully sure if we need it.  perhaps it makes sense to be a bit pragmatic here and introduce the change after we find a use case for it?

Comment (lucacome) [reply client/nginx.go:50]: Removed the timeout for now.

**Pull request merged**